File: paper_artifacts/experiments/llr8/artifacts/sources/kimi27sglang-fortran-skills/fuse_stencil_through_transient__submitted.f90
subroutine fuse_stencil_through_transient_fp64(a, out, LEN_1D, workspace, workspace_size) bind(C, name="fuse_stencil_through_transient_fp64")
  use iso_c_binding
  implicit none
  integer(c_int64_t), value, intent(in) :: LEN_1D
  real(c_double), intent(in) :: a(LEN_1D)
  real(c_double), intent(inout) :: out(LEN_1D)
  type(c_ptr), value, intent(in) :: workspace
  integer(c_int64_t), value, intent(in) :: workspace_size

  integer(c_int64_t) :: i

  !$omp parallel do simd
  do i = 2_c_int64_t, LEN_1D - 2_c_int64_t
     out(i) = (a(i - 1_c_int64_t) + a(i) + a(i + 1_c_int64_t)) &
          & * (a(i) + a(i + 1_c_int64_t) + a(i + 2_c_int64_t))
  end do
  !$omp end parallel do simd
end subroutine fuse_stencil_through_transient_fp64

File: paper_artifacts/experiments/llr9/artifacts/sources/qwen38-c-skills/fuse_stencil_through_transient__submitted.c
#include <stdint.h>
#include <omp.h>

void fuse_stencil_through_transient_fp64(const double *restrict a,
                                         double *restrict out,
                                         int64_t LEN_1D,
                                         uint8_t *workspace,
                                         int64_t workspace_bytes)
{
    (void)workspace;
    (void)workspace_bytes;
    if (LEN_1D < 4) return;
    int64_t hi = LEN_1D - 2; /* i = 1 .. hi-1 */
    #pragma omp parallel for schedule(static)
    for (int64_t i = 1; i < hi; ++i) {
        double s0 = a[i - 1] + a[i] + a[i + 1];
        double s1 = a[i] + a[i + 1] + a[i + 2];
        out[i] = s0 * s1;
    }
}
